src/workers/postUpdated.ts

```diff
@@ -192,29 +192,63 @@ const assignScoutToPost = async ({
   };
 };
 
-const createPost = async ({
-  counter,
-  logger,
+type CheckExistingPostProps = {
+  entityManager: EntityManager;
+  data: Partial<ArticlePost>;
+  counter: opentelemetry.Counter;
+  logger: FastifyBaseLogger;
+  errorMsg: string;
+  id?: string;
+};
+const checkExistingPost = async ({
   entityManager,
   data,
-  submissionId,
-  mergedKeywords,
-  questions,
-}: CreatePostProps): Promise<Post | null> => {
-  const existingPost = await entityManager
+  counter,
+  logger,
+  errorMsg,
+  id,
+}: CheckExistingPostProps): Promise<boolean> => {
+  let builder = entityManager
     .getRepository(Post)
     .createQueryBuilder()
     .select('id')
     .where(
-      'url = :url or url = :canonicalUrl or "canonicalUrl" = :url or "canonicalUrl" = :canonicalUrl',
+      '(url = :url or url = :canonicalUrl or "canonicalUrl" = :url or "canonicalUrl" = :canonicalUrl)',
       { url: data.url, canonicalUrl: data.canonicalUrl },
-    )
-    .getRawOne();
+    );
+  if (id) {
+    builder = builder.andWhere('id != :id', { id });
+  }
```

Comment:
Is `id` not the sole identifier of the post we are trying to update?

Reply:
Ah, I see, you are trying to query if any other post has the canonical/url that we are now updating for this post?

Reply:
I would maybe move this clause in update specifically since when you provide `id` it does not really look for post matching that `id` but looks if any other post matches the url, which to me is not intuitive and opposite to how it works when `id` is not provided for `checkExistingPost` function.

Reply:
Maybe `checkExistingPost` and `checkExistsUrl`?

Reply:
@capJavert Good feedback, the naming was a bit off actually.
I revamped and added tsdoc to better explain the ID.

Let me know if this works?

Reply:
I would still prefer two functions personally but if going this route can we maybe name `id` as `excludeId` so it is immediately obvious that it will exclude?

Reply:
Hmm but it needs to be one query I think?
You want to check all posts that are not the current one to see if it exists right? (maybe i'm misunderstanding you here?)

If so the excludeId is maybe best?

Reply:
> You want to check all posts that are not the current one to see if it exists right

Function does different things, when you don't provide `id` it will tell you if another post exists, but if you provide `id` you are checking if some other post exists aside from the one with `id`.

in short, yeah, I am fine with `excludeId` if two functions complicate it.

Reply:
Ah got you :) 
Yeah felt too much was the same, for now just went with excludeId.

I do think we need to start abstracting both into seperate ones soon anyway.